src/main/java/org/sciborgs1155/robot/drive/Drive.java

```diff
@@ -1,263 +1,181 @@
-import edu.wpi.first.math.util.Units;
-import edu.wpi.first.wpilibj.smartdashboard.Field2d;
-import edu.wpi.first.wpilibj.smartdashboard.FieldObject2d;
-import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.math.kinematics.DifferentialDriveKinematics;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 import io.github.oblarg.oblog.Loggable;
 import io.github.oblarg.oblog.annotations.Log;
+
 import java.util.List;
-import java.util.function.DoubleSupplier;
-import org.photonvision.EstimatedRobotPose;
+import java.util.function.Supplier;
+
 import org.sciborgs1155.lib.failure.Fallible;
-import org.sciborgs1155.lib.failure.FaultBuilder;
 import org.sciborgs1155.lib.failure.HardwareFault;
-import org.sciborgs1155.robot.Constants;
-import org.sciborgs1155.robot.Robot;
+import org.sciborgs1155.robot.Ports.DrivePorts;
 
-public class Drive extends SubsystemBase implements Fallible, Loggable, AutoCloseable {
+import edu.wpi.first.math.kinematics.DifferentialDriveWheelSpeeds;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.motorcontrol.MotorControllerGroup;
 
-  @Log private final ModuleIO frontLeft;
-  @Log private final ModuleIO frontRight;
-  @Log private final ModuleIO rearLeft;
-  @Log private final ModuleIO rearRight;
 
-  private final List<ModuleIO> modules;
+public class Drive extends SubsystemBase implements Loggable, AutoCloseable{
+  //creates StandardDriveMotor
+  StandardDriveMotor driveMotor = new StandardDriveMotor();
+//new formatting, check out StandardDriveMotor for more; just added so it is less messy in this document
+  private final CANSparkMax FRmotor = driveMotor.create(DrivePorts.FR_DRIVE_PORT);
+  private final CANSparkMax FLmotor = driveMotor.create(DrivePorts.FL_DRIVE_PORT);
+  private final CANSparkMax MRmotor = driveMotor.create(DrivePorts.MR_DRIVE_PORT);
+  private final CANSparkMax MLmotor = driveMotor.create(DrivePorts.ML_DRIVE_PORT);
+  private final CANSparkMax BRmotor = driveMotor.create(DrivePorts.BL_DRIVE_PORT);
+  private final CANSparkMax BLmotor = driveMotor.create(DrivePorts.BL_DRIVE_PORT);
 
-  // this should be a generic IMU class, once WPILib implements it
-  @Log private final WPI_PigeonIMU imu = new WPI_PigeonIMU(PIGEON);
+  private final CANSparkMax[] rightSparks = {FRmotor, MRmotor, BRmotor};
+  private final CANSparkMax[] leftSparks = {FLmotor, MLmotor, BLmotor};
 
-  public final SwerveDriveKinematics kinematics = new SwerveDriveKinematics(MODULE_OFFSET);
+  private final MotorControllerGroup rightMotors = new MotorControllerGroup(rightSparks);
+  private final MotorControllerGroup leftMotors = new MotorControllerGroup(leftSparks);
 
-  // Odometry and pose estimation
-  private final SwerveDrivePoseEstimator odometry;
+  // Gyros
+  @Log private final WPI_PigeonIMU gyro = new WPI_PigeonIMU(DrivePorts.GYRO_PORT);
+  //Right and Left side encoders
+  @Log private final RelativeEncoder rightEncoder = FRmotor.getEncoder();
```

Comment:
The `@Log` annotation only works on `Sendable`s, primitives, or `Strings`. Pay attention to where you're using it because `RelativeEncoder`, for example, cannot be logged by Oblog.